### PR TITLE
Use jupyter

### PR DESCRIPTION
**Pull request by bollwyvl**
This builds on #488, adding the Docker changes needed for deployment.

It is now:
- based on `debian/jessie`
- still uses pip
  - Q: should we use conda?
- installs `notebook`, and not just `nbconvert` as we use the static assets from notebook, as well as some ipython util functions
  - i assume the assets will be on npm/bower eventually...
  - i am tempted to install nbconvert master, where most of our upstream fixes will come from, so that we can take advantage of one of the goals of TBS without pulling the bleeding edge of everything

This should touch off a round of ticket closings as we get unbackported nbconvert fixes!

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,37 +1,58 @@
...
+    nodejs \
+    nodejs-legacy \
+    npm \
+    pandoc \
+    python3-dev \
+    python3-pip \
+    sqlite3 \
+    zlib1g-dev \
+  && apt-get clean \
+  && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
```
Comment (rgbkrk) [inline Dockerfile]: Looking good.

File: Dockerfile [hunk 2]
```diff
@@ -1,37 +1,58 @@
...
 
 # To change the number of threads use
 # docker run -d -e NBVIEWER_THREADS=4 -p 80:8080 nbviewer
 ENV NBVIEWER_THREADS 2
 EXPOSE 8080
 
-RUN pip install invoke && \
-    pip3 install invoke
+RUN pip3 install invoke
```
Comment (rgbkrk) [inline Dockerfile]: Yay for one python, the true python: Python 3! :snake:

File: Dockerfile [hunk 3]
```diff
@@ -1,37 +1,58 @@
...
+# front-end dependencies
 ADD ["./nbviewer/static/bower.json", "./nbviewer/static/.bowerrc", \
      "/srv/nbviewer/nbviewer/static/"]
 RUN invoke bower
 
+# build css
 ADD . /srv/nbviewer/
 RUN invoke less
 
+# root up until now!
```
Comment (rgbkrk) [inline Dockerfile]: Yeah, guess we should fix that later.

File: nbviewer/app.py [hunk 4]
```diff
@@ -23,7 +23,7 @@
 
 from jinja2 import Environment, FileSystemLoader
 
-from IPython.config import Config
+from traitlets.config import Config
```
Comment (rgbkrk) [inline nbviewer/app.py]: :boom:

File: nbviewer/static/bower.json [hunk 5]
```diff
@@ -8,7 +8,7 @@
     "moment": "~2.8.4",
     "bootstrap": "components/bootstrap#~3.3",
     "font-awesome": "~4",
-    "pygments": "2.0.0",
-    "reveal.js": "~3.0.0"
+    "pygments": "~2.0.0",
+    "reveal.js": "~3.1.0"
```
Comment (rgbkrk) [inline nbviewer/static/bower.json]: Cool.

File: requirements.txt [hunk 6]
```diff
@@ -1,10 +1,7 @@
-tornado>=3.1.1
-
-pycurl
-pylibmc
+elasticsearch
```
Comment (rgbkrk) [inline requirements.txt]: You don't have to rip it out of here now, but we could toss elasticsearch in a follow on PR (then figure out what we're doing after).

**Comment by rgbkrk** [comment]
Now we wait for Docker Hub!

**Comment by bollwyvl** [comment]
tick tock tick tock.

thanks for the quick, thorough review and merge!

File: nbviewer/static/bower.json (hunk 5 above)
Comment (damianavila) [inline nbviewer/static/bower.json]: yep... now we are on 3.1.0 in both sides (nbviewer and nbconvert) :+1:

**Pull request merged**